code/modules/cargo/blackmarket/blackmarket_market.dm

```diff
@@ -0,0 +1,54 @@
+/datum/blackmarket_market
+	/// Name for the market.
+	var/name = "huh?"
+
+	/// Available shipping methods and prices, just leave the shipping method out that you don't want to have.
+	var/list/shipping
+
+
+	// Automatic vars, do not touch these.
+	/// Items available from this market, populated by SSblackmarket on initialization.
+	var/list/available_items = list()
+	/// Item categories available from this market, only items which are in these categories can be gotten from this market.
+	var/list/categories	= list()
+
+/// Adds item to the available items and add it's category if it is not in categories yet.
+/datum/blackmarket_market/proc/add_item(datum/blackmarket_item/item)
+	if(!prob(initial(item.availability_prob)))
+		return FALSE
+
+	if(ispath(item))
+		item = new item()
+
+	if(!(item.category in categories))
+		categories += item.category
+		available_items[item.category] = list()
+
+	available_items[item.category] += item
+	return TRUE
+
+/// Handles buying the item, this is mainly for future use and moving the code away from the uplink.
+/datum/blackmarket_market/proc/purchase(item, category, method, obj/item/blackmarket_uplink/uplink, user)
+	if(!istype(uplink) || !(method in shipping))
+		return FALSE
+
+	for(var/datum/blackmarket_item/I in available_items[category])
+		if(I.type != item)
+			continue
+		var/price = I.price + shipping[method]
+		// I can't get the price of the item and shipping in a clean way to the UI, so I have to do this.
+		if(uplink.money < price)
+			to_chat("<span class='warning'>You don't have enough credits in [uplink] for [I] with [method] shipping.</span>")
+			return FALSE
+
+		if(I.buy(uplink, user, method))
+			uplink.money -= price
+			return TRUE
+		return FALSE
+
+/datum/blackmarket_market/blackmarket
+	name = "Black Market"
+	shipping = list(SHIPPING_METHOD_LTSRBT	=50,
```

Comment:
LTSRBT should be pretty expensive, considering its completely safe.

Reply:
It is not really that safe, and also the cost is low because it is fairly expensive from cargo.

Reply:
10k from cargo, reasonable delay. Yeah okay then.